proto/v1/rollout_service.proto

```diff
@@ -101,9 +117,406 @@ message Plan {
+    DATABASE_STATEMENT_TYPE_REPORT = 6;
+    DATABASE_STATEMENT_AFFECTED_ROWS_REPORT = 7;
+    DATABASE_CONNECT = 8;
+    DATABASE_GHOST_SYNC = 9;
+    DATABASE_PITR_MYSQL = 10;
+  }
+  Type type = 3;
+
+  enum Status {
+    STATUS_UNSPECIFIED = 0;
+    RUNNING = 1;
+    DONE = 2;
+    FAILED = 3;
+    CANCELED = 4;
+  }
+  Status status = 4;
+
+  // Format: instances/{instance}/databases/{database}
+  string target = 5;
+  // Format: sheets/{sheet}
+  string sheet = 6;
+
+  string detail = 7;
+  repeated Result results = 8;
+
+  message Result {
+    enum Namespace {
+      NAMESPACE_UNSPECIFIED = 0;
+      BYTEBASE = 1;
+      ADVISOR = 2;
+    }
+    Namespace namespace = 1;
+    int64 code = 2;
+    enum Status {
+      STATUS_UNSPECIFIED = 0;
+      ERROR = 1;
+      WARNING = 2;
+      SUCCESS = 3;
+    }
+    Status status = 3;
+    string title = 4;
+    string content = 5;
+    int64 line = 6;
+    string detail = 7;
+  }
+}
+
+message GetRolloutRequest {
+  // The name of the rollout to retrieve.
+  // Format: projects/{project}/rollouts/{rollout}
+  string name = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message Rollout {
+  // The resource name of the rollout.
+  // Format: projects/{project}/rollouts/{rollout}
+  string name = 1;
+
+  // The system-assigned, unique identifier for a resource.
+  string uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The plan that this rollout is based on.
+  // Format: projects/{project}/plans/{plan}
+  string plan = 3;
+
+  string title = 4;
+
+  // stages and thus tasks of the rollout.
+  repeated Stage stages = 5;
+}
+
+message Stage {
+  // Format: projects/{project}/rollouts/{rollout}/stages/{stage}
+  string name = 1;
+  // The system-assigned, unique identifier for a resource.
+  string uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Format: environments/{environment}
+  string environment = 3;
+
+  string title = 4;
+  repeated Task tasks = 5;
+}
+
+message Task {
+  // Format: projects/{project}/rollouts/{rollout}/stages/{stage}/tasks/{task}
+  string name = 1;
+  // The system-assigned, unique identifier for a resource.
+  string uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
   string title = 3;
 
-  string description = 4;
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+    GENERAL = 1;
+    // use payload DatabaseCreate
+    DATABASE_CREATE = 2;
+    // use payload DatabaseSchemaBaseline
+    DATABASE_SCHEMA_BASELINE = 3;
+    // use payload DatabaseSchemaUpdate
+    DATABASE_SCHEMA_UPDATE = 4;
+    // use payload DatabaseSchemaUpdate
+    DATABASE_SCHEMA_UPDATE_SDL = 5;
+    // use payload DatabaseSchemaUpdate
+    DATABASE_SCHEMA_UPDATE_GHOST_SYNC = 6;
+    // use payload nil
+    DATABASE_SCHEMA_UPDATE_GHOST_CUTOVER = 7;
+    // use payload DatabaseDataUpdate
+    DATABASE_DATA_UPDATE = 8;
+    // use payload DatabaseBackup
+    DATABASE_BACKUP = 9;
+    // use payload DatabaseRestoreRestore
+    DATABASE_RESTORE_RESTORE = 10;
```

Comment:
DATABASE_RESTORE and DATABASE_CUTOVER ?

Reply:
or 

DATABASE_RESTORE and DATABASE_PITR_CUTOVER

Just trying avoid DATABASE_RESTORE_RESTORE (just make people think it's a typo)

Reply:
DATABASE_RESTORE_RESTORE and DATABASE_RESTORE_CUTOVER work together to complete DATABASE_RESTORE. 
Maybe we can call DATABASE_RESTORE_RESTORE DATABASE_RESTORE_RECOVERY